### PR TITLE
Add friendly chat codelab

### PR DESCRIPTION
**Pull request by domesticmouse**
Code lab from https://github.com/flutter/friendlychat-steps migrated to null safety, with pedantic and a smoke test

### CODE REVIEW

**Comment by domesticmouse** [comment]
FYI @sfshaza2

**Comment by domesticmouse** [comment]
TBR: @RedBrogdon

File: friendly_chat/step_09/lib/main.dart [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('FriendlyChat'),
+        elevation: Theme.of(context).platform == TargetPlatform.iOS ? 0.0 : 4.0,
+      ),
+      body: Container(
+        decoration: Theme.of(context).platform == TargetPlatform.iOS //new
+            ? BoxDecoration(
+                border: Border(
+                  top: BorderSide(color: Colors.grey[200]!),
```
Comment (amm-catapult) [inline friendly_chat/step_09/lib/main.dart:104]: The `!` null check operator shown here is correct, but it's missing from the [Codelab snippet](https://codelabs.developers.google.com/codelabs/flutter#8).

**Comment by sfshaza2** [comment]
@domesticmouse, it looks like @amm-catapult found a tiny error?

**Comment by domesticmouse** [comment]
@amm-catapult thanks for the bug report. Fixed =)

**Comment by amm-catapult** [comment]
Nice one. Cheers team! :)

**Pull request merged**